### PR TITLE
Added experiments in the experiment folder

### PR DESCRIPTION
**Pull request by syedmohsinbukhari**
### Change Log
----------------------------------------
#### Tensorflow 2.0 MNIST experiment
- Prepare dataset
- Select an optimizer
- Define the loss functions
- Train the neural net
- Test the neural net

#### Backpropagation using numpy
- Added Sigmoid and Relu with their derivatives
- Added SSE Loss
- Added SGD optimizer
- Added FeedForward neural network
- Added an example of how to use all this

### CODE REVIEW

**Comment by Bazarovay** [comment]
Decent work. We have to make sure, however, that we don't end up just copy pasting stuff without understanding it. Nobody wants that...

**Comment by syedmohsinbukhari** [comment]
added another example. please review

**Comment by syedmohsinbukhari** [comment]
added test cases for activation functions today and organized activation functions into a module of their own

**Comment by syedmohsinbukhari** [comment]
started constructing main Model classes with forward and backward functions.

File: experiments/backprop_numpy/activations.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+author: syedmohsinbukhari@googlemail.com
+
+Reference
+http://neuralnetworksanddeeplearning.com/chap2.html
+"""
+
+from numpy import exp
+
+
+class ActivationFunction:
```
Comment (Bazarovay) [inline experiments/backprop_numpy/activations.py:13]: Please add comments

File: experiments/backprop_numpy/models_sequential.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        return
+
+
+if __name__ == '__main__':
+    ffm = FeedForwardModel([3, 2, 1])
+
+    nn_out = ffm.forward(np.array([[1.0], [2.0], [3.0]]), with_grads=True)
+    print(f"Neural Network Output: {nn_out}")
+
+    ffm.backward(np.array([[0.1]]))
```
Comment (Bazarovay) [inline experiments/backprop_numpy/models_sequential.py]: Beautiful

**Comment by Bazarovay** [comment]
Cool stuff. Do you want to merge it?

**Review by Bazarovay — APPROVED** [review]
